applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -123,18 +132,28 @@ LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_LOG_LEVEL);
 		#define MCUBOOT_PRIMARY_SLOT_ID   DT_FIXED_PARTITION_ID(MCUBOOT_PRIMARY_NODE)
 		#define MCUBOOT_SECONDARY_SLOT_ID DT_FIXED_PARTITION_ID(MCUBOOT_SECONDARY_NODE)
 
-		/* Use range check to allow for placing MCUboot header in a separate partition,
-		 * so the application code partition is not an alias for the MCUboot partition,
-		 * but a subpartition of the MCUboot partition.
-		 */
-		#if (CODE_PARTITION_START_ADDR >= MCUBOOT_PRIMARY_START_ADDR) && \
-		    (CODE_PARTITION_START_ADDR < MCUBOOT_PRIMARY_END_ADDR)
-			#define DFU_SLOT_ID MCUBOOT_SECONDARY_SLOT_ID
-		#elif (CODE_PARTITION_START_ADDR >= MCUBOOT_SECONDARY_START_ADDR) && \
-		      (CODE_PARTITION_START_ADDR < MCUBOOT_SECONDARY_END_ADDR)
-			#define DFU_SLOT_ID MCUBOOT_PRIMARY_SLOT_ID
+
+		#if CONFIG_DESKTOP_CONFIG_CHANNEL_DFU_MCUBOOT_RAM_LOAD
+			/* Use the macro alias for the DFU slot ID that needs to be set in runtime
+			 * when the MCUboot is configured for the RAM load mode.
+			 */
+			#define DFU_SLOT_ID_INVALID (0xFF)
```

Comment:
Consider "localizing" implementation for getting DFU slot ID in one place. Maybe we could e.g. create a small function that would rely on either build time or runtime check under the hood?

Reply:
The `dfu_slot_id_runtime_init` is not too heavy, maybe we could simply call it every time we need to access the information (instead of calling it once during boot)